### DISCUSSION TITLE
Understanding llama-bench

### DISCUSSION THREAD

**Discussion by aneeshmb02** [ggml-org/llama.cpp#7195] [Q&A] [answered]
I am trying to understand what data(prompts) is being used by the [llama-bench](https://github.com/ggerganov/llama.cpp/tree/master/examples/llama-bench) tool. 
I actually want to compare the performance of different models with different configurations (varying hardware and params). Hence, I need a way to automate the testing the process. Llama-bench seems to be doing that but I want control over the prompts that are used for benchmarking.

**Comment by slaren** [comment, marked as answer]
The prompts used by `llama-bench` are randomly generated sequences of tokens. You can configure the length, but not the contents.

**Reply by aneeshmb02** [reply]
Thanks a lot!

**Reply by aneeshmb02** [reply]
I tried using [mistral 7b](https://huggingface.co/TheBloke/Mistral-7B-Instruct-v0.2-GGUF) with llama-bench but the output shows llama 8b 
![image](https://github.com/ggerganov/llama.cpp/assets/138118545/ee74767d-e588-482d-aafa-c27113e54431)
Why is that?

**Reply by slaren** [reply]
The heuristic test to detect the model type is wrong and needs to be fixed.

**Reply by aneeshmb02** [reply]
Oh okay

**Comment by kongweiming** [comment]
excuse me ,when tg == 128, what is pp's value this moment?

**Reply by stepinto** [reply]
> excuse me ,when tg == 128, what is pp's value this moment?

Looking into the code, I guess it is 1.  The decode benchmark is not affected by the `-p` parameter.  It always start with a prompt of single random token.  Please correct me if I was wrong.
